modules/@apostrophecms/piece-type/ui/apos/components/AposPiecesManager.vue

```diff
@@ -4,23 +4,34 @@
     @esc="cancel" @no-modal="$emit('safe-close')"
     @inactive="modal.active = false" @show-modal="modal.showModal = true"
   >
-    <template v-if="relationshipField" #primaryControls>
+    <template #secondaryControls>
       <AposButton
+        v-if="relationshipField"
         type="default" label="Cancel"
         @click="cancel"
       />
       <AposButton
-        :label="`Save`" type="primary"
-        :disabled="relationshipErrors === 'min'"
-        @click="saveRelationship"
+        v-else
+        type="default" label="Finished"
+        @click="cancel"
       />
     </template>
-    <template v-else #primaryControls>
+    <template #primaryControls>
+      <!-- TODO Make sure this gets positioned correctly after Vue3/teleport -->
+      <AposContextMenu
+        v-if="relationshipField"
```

Comment:
This needs to be conditional on `relationshipMore.menu.length`

Reply:
generalized the More menu to be independent of the whether the modal has a relationship field .. This should allow other scenarios to add to it should they need to